lib/guest_installation_and_configuration_base.pm

```diff
@@ -1734,15 +1735,18 @@ sub start_guest_installation {
     my $_guest_installation_dryrun_log = "$common_log_folder/$self->{guest_name}/$self->{guest_name}" . "_installation_dryrun_log_" . $_start_installation_timestamp;
     my $_guest_installation_log = "$common_log_folder/$self->{guest_name}/$self->{guest_name}" . "_installation_log_" . $_start_installation_timestamp;
     assert_script_run("touch $_guest_installation_log && chmod 777 $_guest_installation_log");
-    if (script_run("set -o pipefail;$self->{virt_install_command_line_dryrun} 2>&1 | tee -a $_guest_installation_dryrun_log", timeout => 600 / get_var('TIMEOUT_SCALE', 1)) ne 0) {
-        record_info("Guest $self->{guest_name} installation dry run failed", "The virt-install command used is $self->{virt_install_command_line_dryrun}");
-        $self->record_guest_installation_result('FAILED');
+    # Dry run always timeout when downloading initrd from download.opensuse.org in O3
+    my $ret = script_run("set -o pipefail; $self->{virt_install_command_line_dryrun} 2>&1 | tee -a $_guest_installation_dryrun_log", timeout => 600 / get_var('TIMEOUT_SCALE', 1), die_on_timeout => 0);
+    save_screenshot;
+    unless (defined(script_run('set +o pipefail', die_on_timeout => 0))) {
+        reconnect_when_ssh_console_broken;
         script_run("set +o pipefail");
-        save_screenshot;
+    }
+    if ($ret ne 0) {
+        record_info("Guest $self->{guest_name} installation dry run failed", "The virt-install command used is $self->{virt_install_command_line_dryrun}", result => 'fail');
```

Comment:
@Julie-CAO Hi Julie, do I understand correctly that with this commit , you'd like to  mainly add record_info for dry run failure, which in previous code logic it dies directly when timeout arrives?

Reply:
I guess it was not Wayne's intention because

- usually timeout have not happened for SLE
- test is supposed to be continued as there are still other guest need to be installed
- post_fail_hook will not be called when script timeout. It means there are no logs uploaded.

So my change will make test continue and post_fail_hook will be called at the end of the guest installation module.

Reply:
I get the point now. Thanks for explanation. Make sense to me.